client/homebrew/pages/userPage/userPage.jsx

```diff
@@ -36,13 +44,88 @@ const UserPage = createClass({
 	renderBrews : function(brews){
 		if(!brews || !brews.length) return <div className='noBrews'>No Brews.</div>;
 
-		const sortedBrews = _.sortBy(brews, (brew)=>{ return brew.title; });
+		const sortedBrews = this.sortBrews(brews, this.state.sortType);
 
 		return _.map(sortedBrews, (brew, idx)=>{
 			return <BrewItem brew={brew} key={idx}/>;
 		});
 	},
 
+	sortBrews : function(brews, sortType){
+		if(sortType == 'alpha') {
+			return _.orderBy(brews, (brew)=>{ return _.deburr(brew.title.toLowerCase()); }, this.state.sortDir);
+		}
+		if(sortType == 'created'){
+			return _.orderBy(brews, (brew)=>{ return brew.createdAt; }, this.state.sortDir);
+		}
+		if(sortType == 'updated'){
+			return _.orderBy(brews, (brew)=>{ return brew.updatedAt; }, this.state.sortDir);
+		}
+		if(sortType == 'views'){
+			return _.orderBy(brews, (brew)=>{ return brew.views; }, this.state.sortDir);
+		}
+		if(sortType == 'latest'){
+			return _.orderBy(brews, (brew)=>{ return brew.lastViewed; }, this.state.sortDir);
+		}
+		return _.orderBy(brews, (brew)=>{ return _.deburr(brew.title.toLowerCase()); }, this.state.sortDir);
+	},
```

Comment:
Can we make this like a switch statement or something?

Also, if you want the "fallback" sorting to just be alphabetical, we don't need a separate "if-then" check for alphabetical mode. Just let it fall through to the default sorting mode if `sortType == "alpha"`

Reply:
There's a lot of DRY violation going on here. I feel like there should be a better way; I'll look into it further tonight.

Something sort of like this:

```
brewSortingData : function(brew){
    switch(sortType) {
        case 'alpha':
            return _.deburr(brew.title.toLowerCase());
        case [....]
    };
},

sortBrew : function(brews, sortType){
    return _.orderBy(brews, (brew)=>{ brewSortingData(brew) }, this.state.sortDir);
}
```

Something like that, so all the stuff that changes from line to line in the current `sortBrews` function takes place in a separate sub-function `brewSortingData` or some such name, and thus the function is reduced to effectively a single line containing the `_.orderBy`.